sdk/src/mmmClient.ts

```diff
@@ -300,6 +300,74 @@ export class MMMClient {
     transferHookProvider?: TransferHookProvider,
   ): Promise<TransactionInstruction> {
     if (!this.poolData) throw MMMClient.ErrPoolDataEmpty;
+    let builder:
+      | ReturnType<MmmMethodsNamespace['solOcpFulfillBuy']>
+      | ReturnType<MmmMethodsNamespace['solFulfillBuy']>
+      | ReturnType<MmmMethodsNamespace['solMip1FulfillBuy']>
+      | ReturnType<MmmMethodsNamespace['solExtFulfillBuy']>
+      | ReturnType<MmmMethodsNamespace['mplCoreFulfillBuy']>;
+
+    const mintOrCoreAsset = await this.conn.getAccountInfo(assetMint);
+    let { key: buysideSolEscrowAccount } = getMMMBuysideSolEscrowPDA(
+      MMMProgramID,
+      this.poolData.pool,
+    );
+    const { key: sellState } = getMMMSellStatePDA(
+      MMMProgramID,
+      this.poolData.pool,
+      assetMint,
+    );
+    if (!!mintOrCoreAsset && isMplCoreAsset(mintOrCoreAsset)) {
+      const asset = deserializeAssetV1(
```

Comment:
we can build this into the rpc metadata provider right?

Reply:
yeah I plan to do it another PR